rpc/src/v1/helpers/subscribers.rs

```diff
@@ -42,15 +41,36 @@ impl Id {
 	}
 }
 
+#[cfg(not(test))]
+mod random {
+	use rand;
+
+	pub type Rng = rand::rngs::OsRng;
+
+	pub fn new() -> Rng { Rng::new().expect("Valid random source is required.") }
+}
+
+#[cfg(test)]
+mod random {
+	use rand::SeedableRng;
+	use rand_xorshift::XorShiftRng;
+
+	const RNG_SEED: [u8; 16] = [0u8; 16];
+
+	pub type Rng = XorShiftRng;
+
+	pub fn new() -> Rng { Rng::from_seed(RNG_SEED) }
+}
+
 pub struct Subscribers<T> {
-	rand: EntropyRng,
+	rand: random::Rng,
```

Comment:
@tomusdrw could you take a look at this commit, does it look good to you?
Also `can_push_work` test reproducibly fails on my laptop, any hints?

Reply:
Seems sane, can you post the error and maybe some traces (`miner=trace,stratum=trace`) from that test? Might be related to the todo you've left?

Reply:
Here is the output:
```
TRACE 2019-05-29T09:17:15Z: ethcore_stratum: New worker #miner1 registered
TRACE 2019-05-29T09:17:15Z: ethcore_stratum::tests: Received authorization confirmation
TRACE 2019-05-29T09:17:15Z: ethcore_stratum::tests: Pusing work to peers
TRACE 2019-05-29T09:17:15Z: ethcore_stratum: pushing work for 1 workers (payload: '{ "id": 17, "method": "mining.notify", "params": { "00040008", "100500" } }')
TRACE 2019-05-29T09:17:15Z: ethcore_stratum: pusing work to 127.0.0.1:51510
TRACE 2019-05-29T09:17:15Z: ethcore_stratum::tests: Ready to read work from server
TRACE 2019-05-29T09:17:15Z: ethcore_stratum::tests: Received work from server
test tests::can_push_work ... FAILED

failures:

---- tests::can_push_work stdout ----
thread 'tests::can_push_work' panicked at 'assertion failed: `(left == right)`
  left: `"{ \"id\": 17, \"method\": \"mining.notify\", \"params\": { \"00040008\", \"100500\" } }\n"`,
 right: `""`', miner/stratum/src/lib.rs:497:3
```

Reply:
Ah, actually it fails on master as well: https://github.com/paritytech/parity-ethereum/issues/10589, seems to fixed in jsonrpc upstream.